chalice/deploy/newdeployer.py

```diff
@@ -479,29 +524,47 @@ def _do_storevalue(self, instruction):
         # type: (models.StoreValue) -> None
         self.variables[instruction.name] = self.stack[-1]
 
+    def _do_loadvalue(self, instruction):
+        # type: (models.LoadValue) -> None
+        self.stack.append(self.variables[instruction.varname])
+
     def _do_recordresource(self, instruction):
         # type: (models.RecordResource) -> None
-        self.resource_values.append({
+        payload = {
             'name': instruction.resource_name,
             'resource_type': instruction.resource_type,
             instruction.name: self.stack[-1],
-        })
+        }
+        self._add_to_deployed_values(payload)
 
     def _do_recordresourcevariable(self, instruction):
         # type: (models.RecordResourceVariable) -> None
-        self.resource_values.append({
+        payload = {
             'name': instruction.resource_name,
             'resource_type': instruction.resource_type,
             instruction.name: self.variables[instruction.variable_name],
-        })
+        }
+        self._add_to_deployed_values(payload)
 
     def _do_recordresourcevalue(self, instruction):
         # type: (models.RecordResourceValue) -> None
-        self.resource_values.append({
+        payload = {
             'name': instruction.resource_name,
             'resource_type': instruction.resource_type,
             instruction.name: instruction.value,
-        })
+        }
+        self._add_to_deployed_values(payload)
+
+    def _add_to_deployed_values(self, payload):
+        # type: (Dict[str, str]) -> None
+        key = payload['name']
+        if key not in self._resource_value_index:
+            self._resource_value_index[key] = payload
+            self.resource_values.append(payload)
+        else:
+            # If the key already exists, we merge the new payload
+            # with the existing payload.
```

Comment:
If you want to store multiple deployed values for a resource.  If you take a look at this commit I added (https://github.com/aws/chalice/pull/735/commits/5972786a0dfc931c679060cb59ea3b6b5ea871cd) you can see we want to store the role_arn as well as the role_name to avoid having to parse our the name from the arn when we delete a role.